### PR TITLE
Adding CI folder for dp-reporter-client to test new Library pipeline.

### PR DESCRIPTION
**Pull request by outsidecoder**
### What
Part of a series of PR's to create a new pipeline process to test Libraries:
feature/concourse-template-libraries-test1-ci (https://github.com/ONSdigital/dp-ci/pull/821)
feature/concourse-template-libraries-test1-configs

This PR is first of a few iterations to build up the functionality of the new Concourse pipeline process.  Library to test with new the Concourse Library template.

NOTE: This pipeline will be deleted from Concourse in due course as this is only a test.

ci/scripts/build.sh
  Typical build sh minus some functionality, which is commented out.  
ci/scripts/unit.sh
  Typical uni sh minus some functionality, which is commented out.
ci/unit.yml
ci/build.yml
  Typical build yml minus some functionality, which is commented out.

### How to review
Manually review scripts/config files.  Very similar to a new app.

### Who can review
Anyone really.  This is only a test.  The pipeline will be removed after testing.

### CODE REVIEW

File: ci/unit.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+
+platform: linux
+
+image_resource:
+  type: docker-image
+  source:
+    repository: golang
+    tag: 1.13.8
```
Comment (gedge) [inline ci/unit.yml:9]: 1.15?

**Review by gedge — CHANGES_REQUESTED** [review]

File: ci/build.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  type: docker-image
+  source:
+    repository: golang
+    tag: 1.13.8
+
+inputs:
+  - name: dp-reporter-client
+    path: dp-reporter-client
+
+# outputs:
```
[the remark this answers is not in the document]
Comment (outsidecoder) [reply ci/build.yml:15]: I was going to add the functionality in through various PR's, however with hindsight this is bad form.  So I'll rectify thanks :)

File: ci/unit.yml (hunk 1 above)
Comment (outsidecoder) [reply ci/unit.yml:9]: I left as I wasn't sure if all the libraries were tested to 1.15 yet.  But I'll change to 1.15 thanks :)

File: ci/build.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+---
+
+platform: linux
+
+image_resource:
+  type: docker-image
+  source:
+    repository: golang
+    tag: 1.13.8
```
[the remark this answers is not in the document]
Comment (outsidecoder) [reply ci/build.yml:9]: I left as I wasn't sure if all the libraries were tested to 1.15 yet.  But I'll change to 1.15 thanks :)

File: ci/scripts/build.sh [hunk 4]
```diff
@@ -0,0 +1 @@
+make build
```
Comment (gedge) [inline ci/scripts/build.sh:1]: I imagine this is too basic - when run inside CI, it'll need to look more like https://github.com/ONSdigital/dp-dataset-api/blob/develop/ci/scripts/build.sh so it is in the right directory

Same for unit.sh below

**Review by gedge — CHANGES_REQUESTED** [review]

File: Makefile [hunk 5]
```diff
@@ -1,3 +1,24 @@
+MAIN=dp-reporter-client
+
+BUILD=build
+BUILD_ARCH=$(BUILD)/$(GOOS)-$(GOARCH)
+BIN_DIR?=.
+
+PHONY: all
```
Comment (gedge) [inline Makefile:7]: I think several of these phony lines need fixing, thus:
```suggestion
.PHONY: all
```

File: Makefile [hunk 6]
```diff
@@ -1,3 +1,24 @@
...
+all: audit test build
+
+PHONY: audit
+audit:
+	go list -m all | nancy sleuth
```
Comment (gedge) [inline Makefile:12]: I refer to this line later... [note1]

File: Makefile [hunk 7]
```diff
@@ -1,3 +1,24 @@
...
+
+PHONY: build
+build:
+	@mkdir -p $(BUILD_ARCH)/$(BIN_DIR)
+	go build $(LDFLAGS) -o $(BUILD_ARCH)/$(BIN_DIR)/$(MAIN)
```
Comment (gedge) [inline Makefile:17]: struggling with this line:
- `LDFLAGS` is not set, so can probably be removed
- in fact, this doesn't build anything for me, so wondering what `make build` should do in general (if anything) - will speak to you about it [note2]

File: ci/scripts/audit.sh [hunk 8]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash -eux
+
+cwd=$(pwd)
+
+pushd $cwd/dp-reporter-client
+  nancy build && mv build/dp-reporter-client $cwd/build
```
Comment (gedge) [inline ci/scripts/audit.sh:6]: should this nancy command be the same as the line previously referred to in [note1]
or perhaps:
```suggestion
  make audit
```
certainly the `mv` is not needed

File: ci/scripts/build.sh [hunk 9]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash -eux
+
+cwd=$(pwd)
+
+pushd $cwd/dp-reporter-client
+  make build && mv build/dp-reporter-client $cwd/build
+  cp Dockerfile.concourse $cwd/build
```
Comment (gedge) [inline ci/scripts/build.sh:7]: this can probably be simplified a lot, once we speak about [note2]

**Review by gedge — CHANGES_REQUESTED** [review]

File: ci/scripts/audit.sh (hunk 8 above)
Comment (gedge) [reply ci/scripts/audit.sh:6]: sorry, this comment stil stands

File: ci/scripts/build.sh (hunk 9 above)
Comment (gedge) [inline ci/scripts/build.sh:7]:
```suggestion
  make build
```

File: ci/scripts/audit.sh (hunk 8 above)
Comment (outsidecoder) [reply ci/scripts/audit.sh]: Apologies last commit should have included these.  Git slip up.

File: ci/scripts/build.sh (hunk 9 above)
Comment (outsidecoder) [reply ci/scripts/build.sh]: Should be sorted now.

Comment (outsidecoder) [reply ci/scripts/build.sh]: Latest push should have these changes in now.

**Review by gedge — APPROVED** [review]

**Pull request merged**